vms/platformvm/txs/builder/builder.go

```diff
@@ -141,6 +164,23 @@ type ProposalTxBuilder interface {
 		changeAddr ids.ShortID,
 	) (*txs.Tx, error)
 
+	// stakeAmount: amount the delegator stakes
+	// startTime: unix time they start delegating
+	// endTime: unix time they stop delegating
+	// nodeID: ID of the node we are delegating to
+	// rewardAddress: address to send reward to, if applicable
+	// keys: keys providing the staked tokens
+	// changeAddr: address to send change to, if there is any
+	NewAddPermissionlessDelegatorTx(
```

Comment:
This is unused in this PR but is used in the follow-up: https://github.com/ava-labs/avalanchego/pull/2662